sdk/identity/Azure.Identity/tests/InteractiveBrowserCredentialTests.cs

```diff
@@ -104,6 +104,57 @@ public async Task InteractiveBrowserRefreshException()
             await Task.CompletedTask;
         }
 
+        [Test]
+        [NonParallelizable]
+        public async Task InteractiveBrowserValidateSyncWorkaroundCompatSwitch([Values(true, false)]bool appSwitchSet)
+        {
+            string expInnerExMessage = Guid.NewGuid().ToString();
+            string expToken = Guid.NewGuid().ToString();
+            DateTimeOffset expExpiresOn = DateTimeOffset.UtcNow.AddMinutes(5);
+
+            AppContext.SetSwitch("Azure.Identity.DisableInteractiveBrowserThreadpoolExecution", appSwitchSet);
```

Comment:
don't we need to set DisableInteractiveThreadpoolExecutionEnvVar also?

Reply:
I tried setting both. The problem is once the AppContext switch is set there is no unsetting it, you can set it to false but then `AppContext.TryGetSwitch` still successfully reads the switch as false so we don't fall through to using the environment variable.  I could write a long test which sequentially tests nothing set, then the environment variable, then the AppContext switch if you think it's warranted.